server/src/main/java/tabletop/controllers/TournamentController.java

```diff
@@ -0,0 +1,125 @@
+import tabletop.services.TournamentService;
+
+import java.util.*;
+
+@RestController
+@RequestMapping(value = "/tournament")
+public class TournamentController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TournamentController.class);
+
+    private TournamentService tournamentService;
+
+    public TournamentController(TournamentService tournamentService) {
+        this.tournamentService = tournamentService;
+    }
+
+    @RequestMapping(value = "/types", method = RequestMethod.GET)
+    public List<TournamentType> getTournamentTypesList() {
+        List<TournamentType> tournamentTypes = new ArrayList<>();
+        tournamentTypes.addAll(Arrays.asList(TournamentType.values()));
+        return tournamentTypes;
+    }
+
+    @RequestMapping(value = "/finished", method = RequestMethod.GET)
+    public Collection<Tournament> getFinishedTournaments() {
+        return tournamentService.getFinishedTournaments();
+    }
+
+    @RequestMapping(value = "/show/{tournamentid}", method = RequestMethod.GET)
+    public Tournament getTournamentById(@PathVariable("tournamentid") Long tournamentid) {
+        LOGGER.debug("Return tournament with id: {}", tournamentid);
+        return tournamentService.getTournamentById(tournamentid);
+    }
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST, consumes = "application/json")
+    public void createTournament(@RequestBody Tournament tournament) {
+        LOGGER.debug("Request to create tournament for game: {}", tournament.getGameName());
+        tournamentService.addTournament(tournament);
+    }
+
+    @RequestMapping(value = "/update", method = RequestMethod.PUT, consumes = "application/json")
+    public void updateTournament(@RequestBody Tournament tournament) {
+        if (tournament.getId() == null) {
+            createTournament(tournament);
+        } else {
+            LOGGER.debug("Request to update tournament with id: {}", tournament.getId());
+            tournamentService.addTournament(tournament);
+        }
+    }
+
+    @RequestMapping(value = "/delete/{tournamentid}", method = RequestMethod.DELETE)
+    public void deleteTournament(@PathVariable("tournamentid") Long tournamentid) {
+        LOGGER.debug("Request to delete tournament: {}", tournamentid);
+        tournamentService.deleteById(tournamentid);
+    }
+
+    @RequestMapping(value = "/finalresults/{tournamentid}", method = RequestMethod.GET)
+    public Collection<TournamentFinalResult> getFinalResultsForTournament(@PathVariable("tournamentid") Long tournamentid) {
+        return tournamentService.getFinalResultsForTournament(tournamentid);
+    }
+
+    @RequestMapping(value = "/init/{tournamentid}", method = RequestMethod.GET)
+    public Collection<Pair<User>> getInitialRound(@PathVariable("tournamentid") Long tournamentid) {
+        Tournament tournament = tournamentService.getTournamentById(tournamentid);
+        if (tournament == null) {
+            return null;
```

Comment:
also, currently this null check isn't even needed because if the tournament doesn't exist an exception will be thrown